web/src/stream_create_subscribers_data.ts

```diff
@@ -1,43 +1,47 @@
+import assert from "minimalistic-assert";
+
 import {page_params} from "./page_params";
 import * as people from "./people";
+import type {User} from "./people";
 
-let user_id_set;
+let user_id_set: Set<number>;
 
-export function initialize_with_current_user() {
+export function initialize_with_current_user(): void {
     const current_user_id = page_params.user_id;
-    user_id_set = new Set();
+    user_id_set = new Set<number>();
+    assert(current_user_id !== undefined, "Current user's id is undefined");
```

Comment:
It may be worth discussing in a new #frontend thread the idea of using a fake `user_id` like `-1` rather than` undefined` for spectators to avoid the need to od this sort of assert.